pkgs/development/libraries/hnswlib/default.nix

```diff
@@ -0,0 +1,65 @@
+{ lib
+, stdenv
+, fetchFromGitHub
+, cmake
+, python3
+}:
+let
+  python = python3.withPackages(ps: with ps; [
+    numpy
+  ]);
+in
+
+stdenv.mkDerivation (finalAttrs: {
+  pname = "hnswlib";
+  version = "0.7.0";
+
+  src = fetchFromGitHub {
+    owner = "nmslib";
+    repo = "hnswlib";
+    rev = "refs/tags/v${finalAttrs.version}";
+    hash = "sha256-XXz0NIQ5dCGwcX2HtbK5NFTalP0TjLO6ll6TmH3oflI=";
+  };
+
+  nativeBuildInputs = [
+    cmake
+  ] ++ lib.optional finalAttrs.doCheck python;
+
+  preConfigure = lib.optionalString finalAttrs.doCheck ''
+    pushd tests/cpp
+    ${python.interpreter} update_gen_data.py
+    popd
+  '';
+
+  # this is a header only library so that we don't need to build it
+  dontUseCmakeConfigure = !finalAttrs.doCheck;
+
+  dontBuild = !finalAttrs.doCheck;
```

Comment:
why not just set these to `false`?

Reply:
When the `cmakeConfigurePhase` runs with `doCheck = true`, we want it to run `make` in buildPhase with the Makefile that cmake generated. Otherwise, we don't want it to run `make` because it will try to use the [Makefile](https://github.com/nmslib/hnswlib/blob/359b2ba87358224963986f709e593d799064ace6/Makefile#L1) in the repository.

Reply:
So there will be 2 use cases, one where python is included and the check phase runs and one where the check phase doesn't run and python is not enabled.
Maybe we can just put a comment to say that somewhere ?
How about something like (correct me if I'm wrong of course)
```
2 use cases
- for the python module, we want to run the check and the cmake configure phase
- for the c++ only version, cmake doesn't run and it doesn't export a python module.
```

Reply:
Python seems to be used to generate tests.
If you want to test, build test cases with cmake and python, if not, just install it.